app/models/course/enrol_request.rb

```diff
@@ -1,5 +1,15 @@
 # frozen_string_literal: true
 class Course::EnrolRequest < ActiveRecord::Base
+  validate :validate_user_not_in_course, on: :create
+  validates :user_id, uniqueness: { scope: :course_id }, if: :user_id_changed?
```

Comment:
@weiqingtoh does schema validation generate a similar validation ?  Do we need to remove that ?